SnT/preprocessing/process_inputs.py

```diff
@@ -53,17 +53,20 @@ def skimAndAdd(parquet_input, parquet_output, variables_json, keep_all_features,
   print("Loading input variables...")
   with open(variables_json, "r") as f_var:
     input_variables = json.load(f_var)["input_variables"]
+    print(input_variables)
   # We need to recalculate min and max mvaID because some of the inputs files may not be pre-selected and we need to apply the mvaID preselection here
-  input_variables += [var for var in ["Max_mvaID", "Min_mvaID", "lead_isScEtaEB", "lead_isScEtaEE", "sublead_isScEtaEB", "sublead_isScEtaEE"] if var not in input_variables]
+  input_variables += [var for var in ["Max_mvaID", "Min_mvaID", "lead_isScEtaEB", "lead_isScEtaEE", "sublead_isScEtaEB", "sublead_isScEtaEE","fatjet1_particleNet_XbbVsQCD", "fatjet2_particleNet_XbbVsQCD","fatjet3_particleNet_XbbVsQCD", "fatjet4_particleNet_XbbVsQCD","fatjet1_pt", "fatjet2_pt", "fatjet3_pt", "fatjet4_pt","VBF_first_jet_eta", "VBF_second_jet_eta", "VBF_dijet_mass"] if var not in input_variables]
+  
   skimmedVariables = []
-  if not keep_all_features: skimmedVariables = compositeVariables(input_variables, pf_columns)
+  if not keep_all_features: skimmedVariables = compositeVariables(input_variables, pf_columns, df= None)
+  print("Skimmed Variables that already exist in the dataframe...")
```

Comment:
To catch both cases (keeping all features or not):
```suggestion
  print("Variables that exist in the dataframe...")
```